Pcap++/src/PcapDevice.cpp

```diff
@@ -74,15 +74,16 @@ bool IPcapDevice::matchPacketWithFilter(std::string filterAsString, RawPacket* r
 {
 	static std::string curFilter = "";
 	static struct bpf_program prog;
-	if (curFilter != filterAsString)
+	static bool isProgCompiled = false;
+	if ( (curFilter != filterAsString) || !isProgCompiled )
```

Comment:
Maybe instead of adding a static boolean flag we can give `curFilter` a different initial value, for example: `"--"` and then an empty value will be compiled. Please let me know what you think

Reply:
Hey, I propose another solution, keeping in mind your comment and also resolving another bug I've found. Here's the buggy scenario:
PROPER_FILTER = "";
matchPacketWithFilter(PROPER_FILTER , rawPacket); // filter gets compiled, packet gets checked against it and now curFilter == PROPER_FILTER. Function returns true - ok.
matchPacketWithFilter("some wrong filter", rawPacket); // prog gets freed and the new filter compiled but with error, so prog is freed and **curFilter remains as it was curFilter == PROPER_FILTER**. Function returns false - ok but the state of static objects is buggy.
matchPacketWithFilter(PROPER_FILTER , rawPacket); // because curFilter == PROPER_FILTER  prog will not be compiled **while it was freed before**. Function returns false - **not ok.**

Reply:
That's indeed a bug, thanks for providing a solution to it!